flang/lib/Lower/OpenACC.cpp

```diff
@@ -200,9 +246,262 @@ static void genACC(Fortran::lower::AbstractConverter &converter,
                        firOpBuilder.getUnitAttr());
       }
     }
+  }
+}
+
+static void
+genACC(Fortran::lower::AbstractConverter &converter,
+       Fortran::lower::pft::Evaluation &eval,
+       const Fortran::parser::OpenACCBlockConstruct &blockConstruct) {
+  const auto &beginBlockDirective =
+      std::get<Fortran::parser::AccBeginBlockDirective>(blockConstruct.t);
+  const auto &blockDirective =
+      std::get<Fortran::parser::AccBlockDirective>(beginBlockDirective.t);
+
+  auto &firOpBuilder = converter.getFirOpBuilder();
+  auto currentLocation = converter.getCurrentLocation();
+
+  const auto &accClauseList =
+      std::get<Fortran::parser::AccClauseList>(beginBlockDirective.t);
+
+  if (blockDirective.v == llvm::acc::ACCD_parallel) {
+    mlir::Value async;
```

Comment:
This seems to start a really big block of code. Do you want to split it up into smaller chunks?

Reply:
Yes I plan to do that since this will likely to be reused when I'll work on lowering `!$acc parallel loop`.